docs/contribute/plugin/best_practice.md

```diff
@@ -0,0 +1,27 @@
+---
+sidebar_position: 5
+---
+
+# Best Practice 
+
+When developing a WasmEdge plugin, it's important to follow certain best practices to ensure the plugin is robust, reliable, and efficient. Here are some key best practices to follow:
+
+1. **Choosing the Appropriate Programming Language:**
+   WasmEdge plugins can be developed in several languages including C, C++, and Rust. The choice of language depends on the specific requirements of the plugin and the developer's expertise. The C API is recommended for most use cases due to its simplicity and wide support. However, complex plugins might benefit from the enhanced features of C++ or Rust.
+   
+3. **Writing and Compiling the Plugin:**
+   To create a plugin, you need to define a function in C or C++, and declare it with the `extern "C"` keyword to ensure that the function is exported using C-compatible naming conventions. Then, compile the function as a shared library.
+
+4. **Registering the Plugin with WasmEdge Runtime:**
+   After compiling the shared library, you need to register the function with the WasmEdge runtime. 
+
+5. **Testing the Plugin:**
+   Testing is a crucial step in the development process to ensure that the plugin behaves as expected. There are various testing techniques such as unit testing, integration testing, and fuzz testing that can be used. Unit testing involves testing individual functions or units of code, integration testing involves testing the interaction between the plugin and the Wasm program, and fuzz testing involves generating random input to the plugin to test its robustness and resilience.
+   
+6. **Publishing the Plugin:**
```

Comment:
This is too general. Can you describe more details?

Reply:
@alabulei1 do you think this section is okay now?

Reply:
You provide some details, which is good. But I don't think it's ok. As I discussed with you on Discord, can you please take the `wasm-image` plugin as an example?